ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/MiningConfiguration.java

```diff
@@ -213,7 +215,24 @@ public MutableInitValues getMutableInitValues() {
 
   @Value.Immutable
   public interface MutableInitValues {
-    Bytes DEFAULT_EXTRA_DATA = Bytes.EMPTY;
+    // Obtain the raw short version
+    String RAW_SHORT_VERSION = MutableInitValues.class.getPackage().getImplementationVersion();;
+    // Strip everything from the first '-' or '+' onward 
+    // (handles both e.g. "v23.1.1-dev-ac23d311" or "v23.1.1+commit.abcd1234").
+    String SANITIZED_VERSION = RAW_SHORT_VERSION.replaceFirst("[-+].*", "");
```

Comment:
I think it is better to keep the short commit hash, that helps identifying if it is a released version or a development one, and it should always fit in extra-data.

Reply:
Depends on guarantees of your build pipeline 😀

Keeping the commit hash is more informative

Feel free to absorb or modify this PR